client/components/domains/registrant-extra-info/fr-form.jsx

```diff
@@ -40,82 +47,80 @@ class RegistrantExtraInfoForm extends React.PureComponent {
 		countriesList: PropTypes.object.isRequired,
 		isVisible: PropTypes.bool,
 		onSubmit: PropTypes.func,
-		onStateChanged: PropTypes.func, // Just until we can reduxify the contact details
-		countryCode: PropTypes.string,
-		isProbablyOrganization: PropTypes.bool,
 	}
 
 	static defaultProps = {
 		countriesList: { data: [] },
 		isVisible: true,
-		isProbablyOrganization: false,
-		values: {},
 		onSubmit: noop,
-		onStateChanged: noop,
 	}
 
-	componentWillMount() {
+	constructor( props ) {
+		super( props );
+
 		const defaults = {
-			registrantType: this.props.isProbablyOrganization
+			registrantType: this.props.contactDetails.organization
 				? 'organization' : 'individual',
-			countryOfBirth: this.props.countryCode || 'FR',
+			countryOfBirth: this.props.contactDetails.countryCode || 'FR',
 		};
 
-		const values = {
+		this.props.contactDetails.extra = {
 			...defaults,
-			...this.props.values,
+			...this.props.contactDetails.extra
 		};
+	}
 
-		// It's possible for the parent to pass in null and still have a form
-		// that the user is happy with, so we pass one state out
-		// immediatly to to make sure there's something valid in the parent
-		this.props.onStateChange( values );
+	componentDidUpdate( prevProps, prevState ) {
+		const dateOfBirth = this.compileDateOfBirth();
+
+		if ( dateOfBirth && ! isEqual( prevState, this.state ) ) {
```

Comment:
`compileDateofBirth` should be a cheaper likely path already. I'll tweak it a little more such that it assuredly will be.